### PR TITLE
KIALI-987 Add ServiceEntry support into Istio Config

### PR DESCRIPTION
**Pull request by lucasponce**
This PR extends backend to support ServiceEntry objects in Istio Config feature.

### CODE REVIEW

File: kubernetes/istio_details_service.go [hunk 1]
```diff
@@ -209,11 +209,43 @@ func (in *IstioClient) GetGateway(namespace string, gateway string) (IstioObject
 
 	gatewayObject, ok := result.(*Gateway)
 	if !ok {
-		return nil, fmt.Errorf("%s/%s doesn't return a VirtualService object", namespace, gateway)
+		return nil, fmt.Errorf("%s/%s doesn't return a Gateway object", namespace, gateway)
 	}
 	return gatewayObject.DeepCopyIstioObject(), nil
 }
 
+// GetServiceEntries return all Gateways for a given namespace.
```
Comment (jotak) [inline kubernetes/istio_details_service.go]: I guess:
s/Gateways/Service entries

File: services/business/istio_config.go [hunk 2]
```diff
@@ -35,16 +36,19 @@ func (in *IstioConfigService) GetIstioConfig(criteria IstioConfigCriteria) (mode
 		DestinationPolicies: models.DestinationPolicies{},
 		VirtualServices:     models.VirtualServices{},
 		DestinationRules:    models.DestinationRules{},
+		ServiceEntries:      models.ServiceEntries{},
 		Rules:               models.IstioRules{},
 	}
-	var ggErr, rrErr, dpErr, vsErr, drErr, mrErr error
+	var gg, rr, dp, vs, dr, se []kubernetes.IstioObject
```
Comment (jotak) [inline services/business/istio_config.go]: just out of curiosity, why declaring all of them here if they're only used in their goroutine's scope?

File: services/business/istio_validations.go [hunk 3]
```diff
@@ -108,6 +108,8 @@ func (in *IstioValidationsService) GetIstioObjectValidations(namespace string, o
 			istioDetails.DestinationRules = []kubernetes.IstioObject{dr}
 			objectCheckers = []ObjectChecker{noServiceChecker}
 		}
+	case "serviceentries":
+		// Validations on ServiceEntries are not yet in place
```
Comment (jotak) [inline services/business/istio_validations.go]: Should we enter a JIRA right now for it, or is it something that may even not be necessary to implement?

File: services/business/istio_config.go (hunk 2 above)
Comment (lucasponce) [reply services/business/istio_config.go]: Basically to not duplicate a local variable inside the goroutine.
Internally it was using gg, ggErr := ... now it is changed to gg, ggErr = ... without creating the variable which is tested outside the goroutine.
It was missing some errors.

Comment (jotak) [reply services/business/istio_config.go]: ah yes, makes sense!

File: services/business/istio_validations.go (hunk 3 above)
Comment (lucasponce) [reply services/business/istio_validations.go]: I don't think we will extend too much validations for TP perspective.
I guess important ones are in place as validations per serviceentries and gateways need more "business" context to automate if something is good or not.
I wouldn't add them yet, I think.
Also, in the validations side we are putting an eye into Kiali.
Today it started "some" validation but more technical, without context (just to check if there is some missing field).
I think that we would like to integrate what kiali does in the future, and just to use Kiali as presentation + adding additional business context validation that might have sense (weight, missing services, etc).

**Pull request merged**